File: dotnet/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example1/Article.cs
namespace ObserverDesignPattern.Example1;

public class Article
{
    public string Title { get; set; }
    public string Content { get; set; }
    public DateTime PublishedAt { get; set; }

    public Article(string title, string content)
    {
        Title = title;
        Content = content;
        PublishedAt = DateTime.Now;
    }
}

File: java/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example1/src/main/java/behavioral/observer/example1/Article.java
package behavioral.observer.example1;

import java.time.LocalDateTime;

public class Article {
    private final String title;
    private final String content;
    private final LocalDateTime publishedAt;

    public Article(String title, String content) {
        this.title = title;
        this.content = content;
        this.publishedAt = LocalDateTime.now();
    }

    public String getTitle() { return title; }
    public String getContent() { return content; }
    public LocalDateTime getPublishedAt() { return publishedAt; }
}
